google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcConversions.java

```diff
@@ -165,26 +190,276 @@ private BucketInfo bucketInfoDecode(Bucket from) {
     ifNonNull(from.getCreateTime(), timestampCodec::decode, to::setCreateTimeOffsetDateTime);
     ifNonNull(from.getUpdateTime(), timestampCodec::decode, to::setUpdateTimeOffsetDateTime);
     ifNonNull(from.getEncryption(), Bucket.Encryption::getDefaultKmsKey, to::setDefaultKmsKeyName);
-    ifNonNull(from.getWebsite(), Bucket.Website::getMainPageSuffix, to::setIndexPage);
-    ifNonNull(from.getWebsite(), Bucket.Website::getNotFoundPage, to::setNotFoundPage);
     ifNonNull(from.getRpo(), Rpo::valueOf, to::setRpo);
     ifNonNull(from.getStorageClass(), StorageClass::valueOf, to::setStorageClass);
     ifNonNull(from.getVersioning(), Bucket.Versioning::getEnabled, to::setVersioningEnabled);
     ifNonNull(from.getDefaultEventBasedHold(), to::setDefaultEventBasedHold);
     ifNonNull(from.getLabels(), to::setLabels);
-    ifNonNull(from.getBilling(), Bucket.Billing::getRequesterPays, to::setRequesterPays);
+    if (from.getWebsite() != null
+        || from.getWebsite().getMainPageSuffix() != null
+        || from.getWebsite().getNotFoundPage() != null) {
+      to.setIndexPage(from.getWebsite().getMainPageSuffix());
+      to.setNotFoundPage(from.getWebsite().getNotFoundPage());
+    }
```

Comment:
According to the proto definition, both of these properties are non-optional.
https://github.com/googleapis/java-storage/blob/1f14acd53d481f0a290b33cb207c79be8cc2cc1c/proto-google-cloud-storage-v2/src/main/proto/google/storage/v2/storage.proto#L1520-L1533

This if condition can be simplified to:
```
if (from.hasWebsite())
```